GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -737,6 +749,7 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
                     if (this.newlyCreatedFileAndFolderPaths.Contains(gitUpdate.VirtualPath))
                     {
                         string fullPathToFolder = Path.Combine(this.context.Enlistment.WorkingDirectoryRoot, gitUpdate.VirtualPath);
+                        CheckConditionWithRetry(() => !this.context.FileSystem.DirectoryExists(fullPathToFolder), NumberOfRetriesCheckingForDeleted, MillisecondsToSleepBeforeCheckingForDeleted);

```

Comment:
I think this is a good change for this issue.  

If we ever end up needing a more drastic change (e.g. having a lists of tasks to process in the virtual file system hook) this would be a good candidate to move to that list.